kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/DerivedResourcesTest.java

```diff
@@ -195,75 +197,84 @@ record ConditionStruct(ConditionType type,
     private static List<DynamicTest> testsForDir(List<DesiredFn<KafkaProxy, ?>> dependentResources,
                                                  Path testDir)
             throws IOException {
-        var unusedFiles = childFilesMatching(testDir, "*");
-        Path input = testDir.resolve("in-KafkaProxy.yaml");
-        KafkaProxy kafkaProxy = kafkaProxyFromFile(input);
-        assertMinimalMetadataPresent(kafkaProxy);
-        unusedFiles.remove(input);
-        unusedFiles.remove(testDir.resolve("operator-config.yaml"));
-        unusedFiles.removeAll(childFilesMatching(testDir, "in-*"));
-
-        Context<KafkaProxy> context;
         try {
-            context = buildContext(kafkaProxy, testDir);
-        }
-        catch (IOException e) {
-            throw new UncheckedIOException(e);
-        }
-
-        List<DynamicTest> tests = new ArrayList<>();
-
-        var dr = dependentResources.stream()
-                .flatMap(r -> r.invokeDesired(kafkaProxy, context).values().stream().map(x -> Map.entry(r.resourceType(), x)))
-                .collect(Collectors.groupingBy(Map.Entry::getKey))
-                .entrySet()
-                .stream()
-                .collect(Collectors.toMap(Map.Entry::getKey,
-                        e -> e.getValue().stream().map(Map.Entry::getValue).collect(Collectors.toCollection(() -> new TreeSet<>(
-                                Comparator.comparing(hasMetadata -> hasMetadata.getMetadata().getName()))))));
-        for (var entry : dr.entrySet()) {
-            var resourceType = entry.getKey();
-            var actualResources = entry.getValue();
-            for (var actualResource : actualResources) {
-                String kind = resourceType.getSimpleName();
-                String name = actualResource.getMetadata().getName();
-                var expectedFile = testDir.resolve("out-" + kind + "-" + name + ".yaml");
-                tests.add(DynamicTest.dynamicTest(kind + " '" + name + "' should have the same content as " + testDir.relativize(expectedFile),
-                        () -> {
-                            assertThat(Files.exists(expectedFile)).isTrue();
-                            var expected = loadExpected(expectedFile, resourceType);
-                            assertSameYaml(actualResource, expected);
-                            unusedFiles.remove(expectedFile);
-                        }));
+            var unusedFiles = childFilesMatching(testDir, "*");
+            String inFileName = "in-KafkaProxy.yaml";
+            Path input = testDir.resolve(inFileName);
+            KafkaProxy kafkaProxy = kafkaProxyFromFile(input);
+            assertMinimalMetadata(kafkaProxy.getMetadata(), inFileName);
+
+            unusedFiles.remove(input);
+            unusedFiles.remove(testDir.resolve("operator-config.yaml"));
+            unusedFiles.removeAll(childFilesMatching(testDir, "in-*"));
```

Comment:
I'm not following why we apply such a wide file glob and then surgically remove specific files. 

As far as my current understanding goes 
```
var unusedFiles = childFilesMatching(testDir, "out-*");
```
Would give us the same result. What am I missing?

Reply:
Ack lets just leave it as it is.

Reply:
@robobario is correct. I wanted to avoid there being unused files in there because of the risk that people assume (from the presence of a file) that an assertion is being made, which in fact is not being made, because none of the `DerivedResource` classes produce such a resource.